website/src/components/ViewEventDrawer/ViewEventDrawer.jsx

```diff
@@ -0,0 +1,328 @@
+import React, { Component } from 'react';
+import firebase from 'firebase/app';
+import 'firebase/storage';
+import 'firebase/auth';
+import {
+  Drawer, Icon, Input, Button, message, Switch,
+} from 'antd';
+import PropTypes from 'prop-types';
+import { ChatFeed } from 'react-chat-ui';
+import './ViewEventDrawer.css';
+
+class ViewEventDrawer extends Component {
+  static propTypes = {
+    visible: PropTypes.bool.isRequired,
+    onClose: PropTypes.func,
+  };
+
+  static defaultProps = {
+    onClose: () => console.log('no close function passed'),
```

Comment:
Would there be a situation where this is not passed in? Wouldnt it be better to just make it required.

Reply:
there's a props propagation delay for some reason, causes test errors if required